docs/basic_construction.md

```diff
@@ -0,0 +1,155 @@
+
+- **Random Oracles**:
+  - $H_1: ~~$ A random oracle returning $1$ with probability $1/n_p$, used to probabilistically filter candidate tuples. 
+  - $H_2: ~~$ Another random oracle returning $1$ with probability $q$, applied as a final test to determine if a tuple qualifies as a valid proof.
+
+### The Tree Structure
+- Each root $t \in \[d\]$ is treated as the starting node of a tree.
+- The tree has $u$ levels, where $u$ is the length of the proof tuple.
+- At level $i$, nodes represent tuples of the form $(t, s_1, s_2, \ldots, s_i)$, where $t$ is the root, and $s_1, \ldots, s_i$ are elements.
+
+### Tuple Construction via DFS
+- The prover constructs tuples recursively using DFS.
+- At each level $i$, the prover:
+  - Extends the current tuple by adding an element $s$ from $S_p$.
+  - Checks tuple validity using the random oracle $H_1$. A tuple $(t, s_1, \ldots, s_i)$ is valid if $H_1(t, s_1, \ldots, s_i) = 1$.
+  - If the tuple fails the $H_1$ check, it is discarded, and the DFS does not explore its descendants.
+  - Otherwise, the prover recursively calls DFS on the newly extended tuple to continue the construction process.
+- The process continues until a valid tuple of length $u$ is found, or all possibilities are exhausted.
+
+#### Example: how DFS works
+- Choose a root $t \in \[d\]$ and initialize the search with $(t)$.
+  - At each step, extend the current tuple $(t, s_1, \ldots, s_i)$ by trying every element $s \in S_p$.
+  - Check if $H_1(t, s_1, \ldots, s_i) = 1$:
+    - If true, recursively extend the tuple.
+    - If false, backtrack and try the next candidate for $s_i$.
+- Stop when a valid tuple $(t, s_1, \ldots, s_u)$ is found and satisfies $H_2(t, s_1, \ldots, s_u) = 1$.
+
+## The Protocol
+### Parameter Generation
+The parameter generation process for the basic construction is designed to ensure the protocol achieves the desired levels of *security* and *efficiency*.
+The internal parameters of ALBA, such as $u$, $d$, $q$, are computed based on the security parameters, $n_f$, $n_p$ .
+- Security parameters:
+  - $\lambda_{sec}: $ defines the probability that an adversary, who knows at most $n_f$ elements, can construct a valid proof. It controls the protocol's resistance to adversarial attacks by making forgery negligibly likely.
+  - $\lambda_{rel}: $ defines the probability that an honest prover, with knowledge of a sufficiently large set $S_p$, can successfully generate a valid proof. It ensures the practicality of the protocol for honest participants, emphasizing the importance of feasibility in proof construction.
+- Relationship between the parameters:
+  - The proof length $u$ is directly proportional to the security parameters and inversely proportional to the logarithm of the ratio $n_p / n_f$. 
+    - This means that as the prover's set size $n_p$ grows compared to the threshold $n_f$, the required proof length decreases logarithmically, reflecting the reduced adversarial advantage.
+      $$
```

Comment:
this doesn't render on github

Reply:
When the double dollar sign is in the bullet list, it is not rendered somehow!
So, I put all 3 formulas at the bottom.